### PR TITLE
replaced custom config with confmgr

### LINKED ISSUE #87: Use confmgr
We could improve the ENV and settings management using [cnfmgr](https://gitlab.com/chevdor/confmgr).

### LINKED ISSUE #101: Add ENV for the BaseURL
The [BaseURL](https://github.com/paritytech/substrate-matrix-faucet/blob/a64611292b547f08a519fe4def40eba5ab048704/src/bot/index.ts#L67) should not be hardcoded.

### PR DESCRIPTION
**Pull request by mordamax**
Fix #87
Fix #101

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -13,6 +13,6 @@ jobs:
       - name: Use Node.js
         uses: actions/setup-node@v1
         with:
-          node-version: '14.x'
-      - run: yarn
+          node-version: '16.x'
```
Comment (alvicsam) [inline .github/workflows/lint.yml:16]: I think this GHA can be removed, since the gitlab pipeline has the same job

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -59,23 +59,34 @@ check-typing:
   stage:                           quality
   <<:                              *build
   script:
-    - yarn install
+    - apk add --no-cache git openssh
```
Comment (alvicsam) [inline .gitlab-ci.yml:62]:
```suggestion
    # TODO: remove after https://github.com/paritytech/ci_cd/issues/389
    - apk add --no-cache git openssh
```

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -59,23 +59,34 @@ check-typing:
...
+    - yarn --immutable --network-concurrency 1
     - yarn typing
 
 check-linting:
   stage:                           quality
   <<:                              *build
   script:
-    - yarn install
+    - apk add --no-cache git openssh
```
Comment (alvicsam) [inline .gitlab-ci.yml:70]:
```suggestion
    # TODO: remove after https://github.com/paritytech/ci_cd/issues/389
    - apk add --no-cache git openssh
```

File: .gitlab-ci.yml [hunk 4]
```diff
@@ -59,23 +59,34 @@ check-typing:
...
+    - yarn --immutable --network-concurrency 1
     - yarn lint
 
 check-formatting:
   stage:                           quality
   <<:                              *build
   script:
-    - yarn install
+    - apk add --no-cache git openssh
```
Comment (alvicsam) [inline .gitlab-ci.yml:78]:
```suggestion
    # TODO: remove after https://github.com/paritytech/ci_cd/issues/389
    - apk add --no-cache git openssh
```

File: .gitlab-ci.yml [hunk 5]
```diff
@@ -59,23 +59,34 @@ check-typing:
...
+    - yarn --immutable --network-concurrency 1
     - yarn format
 
+check-build-bot:
+  stage:                           quality
+  <<:                              *build
+  script:
+    - apk add --no-cache git openssh
```
Comment (alvicsam) [inline .gitlab-ci.yml:86]:
```suggestion
    # TODO: remove after https://github.com/paritytech/ci_cd/issues/389
    - apk add --no-cache git openssh
```

File: .github/workflows/lint.yml (hunk 1 above)
Comment (mordamax) [reply .github/workflows/lint.yml]: Good point, removed

**Review by alvicsam — APPROVED** [review]
CI part looks good

**Pull request merged**